### PR TITLE
Update softmax_int8_int16.cc

### PR DESCRIPTION
**Pull request by psharath1**
BUG= Softmax Fix. Reverted changes to the file which introduced failures

### CODE REVIEW

File: tensorflow/lite/micro/kernels/xtensa/softmax_int8_int16.cc [hunk 1]
```diff
@@ -108,11 +108,11 @@ void* XtensaInitSoftmax(TfLiteContext* context, const char* buffer,
 TfLiteStatus XtensaPrepareSoftmax(TfLiteContext* context, TfLiteNode* node) {
 #if defined(HIFI4) || defined(HIFI4_INTERNAL) || defined(HIFI5)
   return PrepareHifi(context, node);
-#elif defined(VISION_P6)
-  TF_LITE_ENSURE_OK(context, SoftmaxPrepareVision(context, node));
-  return kTfLiteOk;
 #else
```
Comment (rascani) [inline tensorflow/lite/micro/kernels/xtensa/softmax_int8_int16.cc:111]: I would suggest creating some consistency between the prepare functions. PrepareHifi calls SoftmaxPrepare, but SoftmaxPrepareVision does not. Consider having either both call SoftmaxPrepare or neither.

File: tensorflow/lite/micro/kernels/xtensa/softmax_int8_int16.cc [hunk 2]
```diff
@@ -108,11 +108,11 @@ void* XtensaInitSoftmax(TfLiteContext* context, const char* buffer,
...
   TF_LITE_ENSURE_OK(context, SoftmaxPrepare(context, node));
+#if defined(VISION_P6)
+  TF_LITE_ENSURE_OK(context, SoftmaxPrepareVision(context, node));
```
Comment (rascani) [inline tensorflow/lite/micro/kernels/xtensa/softmax_int8_int16.cc:114]: This can just be `return SoftmaxPrepareVision(context, node);`

**Review by rascani — APPROVED** [review]

File: tensorflow/lite/micro/kernels/xtensa/softmax_int8_int16.cc (hunk 1 above)
Comment (psharath1) [reply tensorflow/lite/micro/kernels/xtensa/softmax_int8_int16.cc:111]: For Vision P6 we have always followed this method of calling regular prepare followed by prepareVision for all ported kernels.  will need to check with HiFi team

File: tensorflow/lite/micro/kernels/xtensa/softmax_int8_int16.cc (hunk 2 above)
Comment (psharath1) [reply tensorflow/lite/micro/kernels/xtensa/softmax_int8_int16.cc:114]: Will look at incorporating this for the next submissions

**Review by tingyan19 — APPROVED** [review]
